jms/src/main/java/com/sun/ts/tests/jms/core/appclient/closedQueueReceiver/ClosedQueueReceiverTestsAppclientTest.java

```diff
@@ -0,0 +1,135 @@
+        closedQueueReceiver_ejb_vehicle_client: jar.sun-application-client.xml,META-INF/application-client.xml
+        closedQueueReceiver_ejb_vehicle_ejb: META-INF/ejb-jar.xml,jar.sun-ejb-jar.xml
+        closedQueueReceiver_jsp_vehicle: 
+        closedQueueReceiver_jsp_vehicle_web: WEB-INF/web.xml,war.sun-web.xml
+        closedQueueReceiver_servlet_vehicle: 
+        closedQueueReceiver_servlet_vehicle_web: WEB-INF/web.xml,war.sun-web.xml
+
+        Found Descriptors:
+        Client:
+
+        /com/sun/ts/tests/jms/core/appclient/closedQueueReceiver/appclient_vehicle_client.xml
+        /com/sun/ts/tests/common/vehicle/appclient/appclient_vehicle_client.xml
+        Ear:
+
+        */
+        @TargetsContainer("tck-appclient")
+        @OverProtocol("appclient")
+        @Deployment(name = VEHICLE_ARCHIVE, order = 2)
+        public static EnterpriseArchive createDeploymentVehicle(@ArquillianResource TestArchiveProcessor archiveProcessor) {
+        // Client
+            // the jar with the correct archive name
+            JavaArchive closedQueueReceiver_appclient_vehicle_client = ShrinkWrap.create(JavaArchive.class, "closedQueueReceiver_appclient_vehicle_client.jar");
+            // The class files
+            closedQueueReceiver_appclient_vehicle_client.addClasses(
+            com.sun.ts.tests.jms.common.JmsTool.class,
+            com.sun.ts.tests.jms.core.appclient.closedQueueReceiver.ClosedQueueReceiverTests.class,
+            com.sun.ts.tests.common.vehicle.VehicleRunnable.class,
+            com.sun.ts.tests.common.vehicle.VehicleRunnerFactory.class,
+            com.sun.ts.lib.harness.EETest.Fault.class,
+            com.sun.ts.tests.common.vehicle.EmptyVehicleRunner.class,
+            com.sun.ts.lib.harness.EETest.class,
+            com.sun.ts.lib.harness.ServiceEETest.class,
+            com.sun.ts.lib.harness.EETest.SetupException.class,
+            com.sun.ts.tests.common.vehicle.VehicleClient.class
+            );
+            closedQueueReceiver_appclient_vehicle_client.addClass("com.sun.ts.tests.jms.core.appclient.closedQueueReceiver.ClosedQueueReceiverTests$1.class");
+            closedQueueReceiver_appclient_vehicle_client.addClass("com.sun.ts.tests.jms.core.appclient.closedQueueReceiver.ClosedQueueReceiverTests$1.class");
+
+            // The application-client.xml descriptor
+            URL resURL = ClosedQueueReceiverTests.class.getResource("/com/sun/ts/tests/common/vehicle/appclient/appclient_vehicle_client.xml");
+            if(resURL != null) {
+              closedQueueReceiver_appclient_vehicle_client.addAsManifestResource(resURL, "application-client.xml");
+            }
+            // The sun-application-client.xml file need to be added or should this be in in the vendor Arquillian extension?
+            resURL = ClosedQueueReceiverTests.class.getResource("//com/sun/ts/tests/common/vehicle/appclient/appclient_vehicle_client.jar.sun-application-client.xml");
```

Comment:
:eyes:

Reply:
We decided to add the sun descriptors as a flag that a test archive might need a vendor descriptor.

Reply:
Unclear why this one (and few more in other places) needs two slashes at start.